nuxeo-features/nuxeo-platform-oauth/src/main/resources/web/nuxeo.war/oauth2Grant.jsp

```diff
@@ -141,8 +147,21 @@ String logoUrl = LoginScreenHelper.getValueWithDefault(screenConfig.getLogoUrl()
       <fmt:param value="<%=productName%>"/>
     </fmt:message>
 
-    <input name="authorization_key" type="hidden" value="${authorization_key}"/>
-    <input name="state" type="hidden" value="${state}"/>
+    <input name="response_type" type="hidden" value="${response_type}"/>
+    <input name="client_id" type="hidden" value="${client_id}"/>
+    <% if (redirectURI != null) { %>
+    <input name="redirect_uri" type="hidden" value="<%= redirectURI %>"/>
```

Comment:
All values coming from the user like `redirectURI` must be escaped when re-inserted in the HTML. Use `fn:escapeXml`. See login.jsp for example.